plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -910,6 +907,47 @@ private void handleException(FunctionName functionName, String taskId, Exception
         mlTaskManager.updateMLTask(taskId, updated, TIMEOUT_IN_MILLIS, true);
     }
 
+    public synchronized void updateModelCache(String modelId, boolean isPredictorUpdate, ActionListener<String> listener) {
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            ActionListener<String> wrappedListener = ActionListener.runBefore(listener, context::restore);
+            getModel(modelId, ActionListener.wrap(mlModel -> {
+                if (isPredictorUpdate) {
+                    assert FunctionName.REMOTE == mlModel.getAlgorithm()
+                        : "In-place update is only supported on REMOTE models at this time.";
```

Comment:
As we rename the variables to model cache update, let's reflect the logs accordingly. This is applicable for all the `in-place update related message`

However is customer going to see this message? If that's the case, let's make it more customer friendly like:

`Remote models can be updated without undeploying. For other models, please undeploy before updating.`

Reply:
This line is removed in next pr, but nice message